### PR TITLE
Implement js2 w1 tasks for currency conversion

### PR DESCRIPTION
**Pull request by GizemSavci**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    return currencyData;
+}
+
+
+function convertCurrency(){
+    let mainCurrency = document.getElementById('mainCurrency').value;
+    let targetCurrency = document.getElementById('targetCurrency').value;
+    let amount = parseFloat(document.getElementById('amount').value);
+
+    if (isNaN(amount)){
```
Comment (BrantalikP) [inline main.js:34]: I like how you're managing the error cases! Alternatively, you could set the `input` field as 'required'.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        document.getElementById('result').innerText = 'Please enter a valid amount';
+        return;
+    }
+
+    const targetCurrencyData = currencyData.find(currency => currency.conjugate === targetCurrency);
+
+    if (targetCurrencyData){
```
Comment (BrantalikP) [inline main.js:41]: Nice! It would be a nice touch also if you show some message in case the`targetCurrencyData` is null

**Review by BrantalikP — APPROVED** [review]
@GizemSavci Nice job! The code is solid and clean. Also, It’s a strong base for the upcoming assignments!

File: main.js (hunk 1 above)
Comment (GizemSavci) [reply main.js:34]: Okay, that is true. I am going to add 'required' attribute to the input elements.
```
      <input type="text" value="DKK" id="baseCurrency" required/>
      <input type="text" id="conjugateCurrency" required/>
      <input type="number" id="rate" required/>
      <button type="button" onclick="addCurrency()">Add Currency</button>
    </form>
    <form id="convertionForm">
      <input type="text" value="DKK" id="mainCurrency" required/>
      <input type="text" id="targetCurrency" required/>
      <input type="number" id="amount" required/>
```
Changed the button structure.

File: main.js (hunk 2 above)
Comment (GizemSavci) [reply main.js:41]:
```
    const targetCurrencyData = currencyData.find(currency => currency.conjugate === targetCurrency);
    const result = amount*targetCurrencyData.rate;
    if (targetCurrencyData){
        document.getElementById('result').innerText =`${amount} ${targetCurrency} equals ${result.toFixed(2)} ${mainCurrency}`;
        return result;
    }else if(!targetCurrencyData){
      document.getElementById('result').innerText = 'Please try another currency.';
      return result;
    }
}
```
updated with that snippet.

**Pull request merged**